### PR TITLE
Add metadata field to Azure dynamic roles

### PR DESCRIPTION
**Pull request by Zlaticanin**
This PR adds `metadata` field to dynamic roles, similarly to how it's done for Azure static roles. User can set `metadata` on role config and on the request. If a key exists in both, the value from the role's metadata takes precedence.

### CODE REVIEW

File: internal/vault/secrets/ephemeral/azure_access_credentials.go [hunk 1]
```diff
@@ -168,6 +171,16 @@ func (r *AzureAccessCredentialsEphemeralResource) Schema(_ context.Context, _ ep
 				MarkdownDescription: "True if the duration of this lease can be extended through renewal.",
 				Computed:            true,
 			},
+			fieldRequestMetadata: schema.MapAttribute{
```
Comment (siyer-corp) [inline internal/vault/secrets/ephemeral/azure_access_credentials.go:174]: Any reason why this field isn't referring to the consts?

File: vault/resource_azure_secret_backend_role_test.go [hunk 2]
```diff
@@ -276,6 +289,10 @@ resource "vault_azure_secret_backend_role" "test_azure_roles" {
  description	  = "Test for Vault Provider"
  sign_in_audience = "AzureADMyOrg"
  tags             = ["team:engineering"]
+ metadata = {
```
Comment (siyer-corp) [inline vault/resource_azure_secret_backend_role_test.go:383]: Do we need tests for metadata removal, empty maps, and adding metadata to existing roles?

File: internal/vault/secrets/ephemeral/azure_access_credentials_test.go [hunk 3]
```diff
@@ -78,6 +79,15 @@ func TestAccAzureAccessCredentialsEphemeralResource_basic(t *testing.T) {
 							statecheck.ExpectKnownValue("echo.test_azure",
 								tfjsonpath.New("data").AtMapKey("lease_renewable"),
 								knownvalue.NotNull()),
+							statecheck.ExpectKnownValue("echo.test_azure",
```
Comment (siyer-corp) [inline internal/vault/secrets/ephemeral/azure_access_credentials_test.go:82]: If the role does not have any metadata, will the response just have the metadata from the access_credentials? Should we have a test for that scenario?

File: internal/vault/secrets/ephemeral/azure_access_credentials.go [hunk 4]
```diff
@@ -199,6 +212,22 @@ func (r *AzureAccessCredentialsEphemeralResource) Open(ctx context.Context, req
 	role := data.Role.ValueString()
 	credsPath := backend + "/creds/" + role
 
+	var readData map[string][]string
```
Comment (siyer-corp) [inline internal/vault/secrets/ephemeral/azure_access_credentials.go:215]: Would it be good to add documentation for this logic?

File: internal/vault/secrets/ephemeral/azure_access_credentials.go (hunk 1 above)
Comment (Zlaticanin) [reply internal/vault/secrets/ephemeral/azure_access_credentials.go:174]: It is a const, just in the file, since this field was added because of the limitation of having two same fields both in role config and credentials

File: internal/vault/secrets/ephemeral/azure_access_credentials_test.go (hunk 3 above)
Comment (Zlaticanin) [reply internal/vault/secrets/ephemeral/azure_access_credentials_test.go:82]: Yes, if they are set. We do have a test for it?
